src/actions/exportToCSV/ExportToCSVAction.js

```diff
@@ -0,0 +1,130 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2023, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+import { OBJECT_TYPES } from "../../const";
+import {idToQualifiedName} from "../../utils";
+import {saveAs} from 'saveAs';
+
+const SUPPORTED_TYPES = [OBJECT_TYPES.TELEMETRY_OBJECT_TYPE, OBJECT_TYPES.AGGREGATE_TELEMETRY_TYPE];
+
+export default class ExportToCSVAction {
+    constructor(openmct, url, instance) {
+        this.name = 'Export to CSV';
+        this.key = 'exportToCSV';
+        this.cssClass = "icon-export";
+        this.description = 'Export the values for this telemetry object for a given time range.';
+        this.group = 'action';
+        this.priority = 2;
+
+        this.url = url;
+        this.instance = instance;
+        this._openmct = openmct;
+    }
+
+    // Only allow export to CSV for telemetry objects, aggregates and objects with composition.
+    // Objects with composition may not have valid children, which could result in an error.
+    appliesTo(objectPath) {
+        if (!objectPath.length) {
+            return false;
+        }
+
+        const object = objectPath[0];
+        const hasChildren = object.composition && object.composition.length;
```

Comment:
Composition cannot be determined this way. You have to use the composition API.

The composition API will allow you to determine whether or not composition is supported by the object, but you will not be able to get a count synchronously, we will just have to live with that.